source/draw/RasterPacked.ooc

```diff
@@ -31,18 +31,18 @@ RasterPackedCanvas: abstract class extends RasterCanvas {
 	target ::= this _target as RasterPacked
 	init: func (image: RasterPacked) { super(image) }
 	_resizePacked: func <T> (sourceBuffer: T*, source: RasterPacked, sourceBox, resultBox: IntBox2D) {
-		if (this target size == source size && this target stride == source stride && sourceBox == resultBox && sourceBox size == source size && sourceBox leftTop x == 0 && sourceBox leftTop y == 0)
+		if (this target size == source size && this target stride == source stride && sourceBox == resultBox && sourceBox size == source size && sourceBox leftTop x == 0 && sourceBox leftTop y == 0 && source coordinateSystem == this target coordinateSystem)
 			memcpy(this target buffer pointer, sourceBuffer, this target stride * this target height)
 		else if (this interpolationMode == InterpolationMode Fast)
-			This _resizeNearestNeighbour(sourceBuffer, this target buffer pointer as T*, sourceBox, resultBox, source stride, this target stride, this target bytesPerPixel)
+			This _resizeNearestNeighbour(sourceBuffer, this target buffer pointer as T*, source, this target, sourceBox, resultBox)
 		else
-			This _resizeBilinear(sourceBuffer, this target buffer pointer as T*, sourceBox, resultBox, source stride, this target stride, this target bytesPerPixel)
+			This _resizeBilinear(sourceBuffer, this target buffer pointer as T*, source, this target, sourceBox, resultBox)
 	}
-	_resizeNearestNeighbour: static func <T> (sourceBuffer, resultBuffer: T*, sourceBox, resultBox: IntBox2D, sourceStride, resultStride, bytesPerPixel: Int) {
+	_resizeNearestNeighbour: static func <T> (sourceBuffer, resultBuffer: T*, source, target: RasterPacked, sourceBox, resultBox: IntBox2D) {
```

Comment:
Still need to pass the buffer as separate arguments, as rock needs to deduce the `T` implicitly.